tidy3d/plugins/design/design.py

```diff
@@ -128,52 +322,175 @@ def run_batch(
-        -------
-        :class:`.Result`
-            Object containing the results of the design space exploration.
-            Can be converted to ``pandas.DataFrame`` with ``.to_dataframe()``.
+        new_self = self.updated_copy(path_dir=path_dir)
+        result = new_self.run(fn=fn_pre, fn_post=fn_post)
+
+        return result
+
+    def _get_run_count(self) -> int:
+        # For MonteCarlo, Random, RandomCustom
+        if hasattr(self.method, "num_points"):
+            num_points = self.method.num_points
+
+        # For Grid
+        elif isinstance(self.method, MethodGrid):
+            num_points = len(MethodGrid.sample(self.parameters))
+
+        # For BayOpt
+        elif isinstance(self.method, MethodBayOpt):
+            num_points = self.method.initial_iter + self.method.n_iter
+
+        # For GenAlg
+        elif isinstance(self.method, MethodGenAlg):
+            num_points = self.method.solutions_per_pop * self.method.n_generations
+
+        # For ParticleSwarm
+        elif isinstance(self.method, MethodParticleSwarm):
+            num_points = self.method.n_particles * self.method.n_iter
+
+        return num_points
+
+    def estimate_cost(self, fn_pre: Callable) -> float:
         """
+        Compute the maximum FlexCredit charge for the entire design space computation.
+        Require a pre function that should return a Simulation object, Batch object, or collection of either.
+        The pre function is called to estimate the cost - complicated pre functions may cause long runtimes.
+        """
+        # Get output fn_pre for paramters at the lowest span / default
+        arg_dict = {}
+        for param in self.parameters:
+            if hasattr(param, "span"):
+                arg_dict[param.name] = param.span[0]
+            else:  # For ParameterAny objects
+                arg_dict[param.name] = param.allowed_values[0]
 
-        # run the functions using the `method.run_batch`
-        fn_args, fn_values, task_ids, batch_data = self.method.run_batch(
-            parameters=self.parameters,
-            fn_pre=fn_pre,
-            fn_post=fn_post,
-            path_dir=path_dir,
-            **batch_kwargs,
-        )
+        # Compute fn_pre
+        pre_out = fn_pre(**arg_dict)
 
-        # store the pre and post functions in a single source code
-        fn_source_pre = self.get_fn_source(fn_pre)
-        fn_source_post = self.get_fn_source(fn_post)
-        fn_source = self._make_batch_fn_source(fn_source_pre, fn_source_post)
+        def _estimate_sim_cost(sim):
+            job = Job(simulation=sim, task_name="estimate_cost")
 
-        # package the result
-        return self._package_run_results(
-            fn_args=fn_args,
-            fn_values=fn_values,
-            fn_source=fn_source,
-            task_ids=task_ids,
-            batch_data=batch_data,
+            estimate = job.estimate_cost()
+            job.delete()  # Deleted as only a test with initial parameters
```

Comment:
there's gotta be a better way to do this.. there should probably be a `web.estimate_cost(sim)` that does something similar but just doesn't retain the task.

Reply:
maybe we should file a ticket to work on it eventually

Reply:
Yeah it does feel like estimate_cost could have an arg to delete on completion